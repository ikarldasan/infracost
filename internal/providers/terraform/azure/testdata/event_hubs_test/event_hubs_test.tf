provider "azurerm" {
  skip_provider_registration = true
  features {}
}

resource "azurerm_resource_group" "example" {
  name     = "example-notificationhub-resources"
  location = "eastus"
}
resource "azurerm_eventhub_namespace" "standard" {
  name                = "acceptanceTestEventHubNamespace"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  sku                 = "Standard"
  capacity            = 1

  tags = {
    environment = "Production"
  }
}

resource "azurerm_eventhub_namespace" "basic" {
  name                = "acceptanceTestEventHubNamespace"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  sku                 = "Basic"
  capacity            = 1

  tags = {
    environment = "Production"
  }
}

resource "azurerm_eventhub_namespace" "standardwithoutusage" {
  name                = "acceptanceTestEventHubNamespace"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  sku                 = "Standard"
  capacity            = 1

  tags = {
    environment = "Production"
  }
}

resource "azurerm_eventhub_namespace" "basicwithoutusage" {
  name                = "acceptanceTestEventHubNamespace"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  sku                 = "Basic"
  capacity            = 1

  tags = {
    environment = "Production"
  }
}