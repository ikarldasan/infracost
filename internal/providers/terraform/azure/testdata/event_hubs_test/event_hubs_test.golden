
 Name                                                     Monthly Qty  Unit                  Monthly Cost 
                                                                                                          
 azurerm_eventhub_namespace.basic                                                                         
 ├─ Ingress events (Basic)                                          1  1M events                    $0.03 
 └─ Throughput unit (1 MB/s ingress, 2 MB/s egress)                 1  hour                         $0.02 
                                                                                                          
 azurerm_eventhub_namespace.basicwithoutusage                                                             
 ├─ Ingress events (Basic)                           Monthly cost depends on usage: $0.03 per 1M events   
 └─ Throughput unit (1 MB/s ingress, 2 MB/s egress)                 1  hour                         $0.02 
                                                                                                          
 azurerm_eventhub_namespace.standard                                                                      
 ├─ Ingress events (Standard)                                      10  1M events                    $0.28 
 ├─ Throughput unit (1 MB/s ingress, 2 MB/s egress)                 1  hour                         $0.03 
 ├─ Kafka endpoint                                                  1  hour                         $0.09 
 └─ Capture                                                         1  hour                         $0.10 
                                                                                                          
 azurerm_eventhub_namespace.standardwithoutusage                                                          
 ├─ Ingress events (Standard)                        Monthly cost depends on usage: $0.03 per 1M events   
 ├─ Throughput unit (1 MB/s ingress, 2 MB/s egress)                 1  hour                         $0.03 
 ├─ Kafka endpoint                                                  1  hour                         $0.09 
 └─ Capture                                                         1  hour                         $0.10 
                                                                                                          
 PROJECT TOTAL                                                                                      $0.78 

----------------------------------
To estimate usage-based resources use --usage-file, see https://infracost.io/usage-file